### ISSUE TITLE
Errors in Demo Dashboards

### ISSUE THREAD

**Issue by kkalyan** [apache/incubator-superset#4081]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
{"GIT_SHA": "1e79e9cd2a064c8a6daa77bb74e549e0b12cf840", "version": "0.23.0dev"}

#### Births Dashboard 

![image](https://user-images.githubusercontent.com/391567/34122391-2b59a400-e3e1-11e7-9646-eee62e3f5ad5.png)
```
Traceback (most recent call last):
  File "/Users/kkalyan/try/incubator-superset/superset/viz.py", line 265, in get_payload
    rowcount = len(df.index)
AttributeError: 'bool' object has no attribute 'index'
```

#### Misc Charts
![image](https://user-images.githubusercontent.com/391567/34122441-4b50840e-e3e1-11e7-896b-f14383099d21.png)

```
Traceback (most recent call last):
  File "/Users/kkalyan/try/incubator-superset/superset/viz.py", line 262, in get_payload
    df = self.get_df()
  File "/Users/kkalyan/try/incubator-superset/superset/viz.py", line 87, in get_df
    query_obj = self.query_obj()
  File "/Users/kkalyan/try/incubator-superset/superset/viz.py", line 1450, in query_obj
    qry = super(CountryMapViz, self).query_obj()
  File "/Users/kkalyan/try/incubator-superset/superset/viz.py", line 173, in query_obj
    from_dttm = utils.parse_human_datetime(since) - self.time_shift
TypeError: unsupported operand type(s) for -: 'NoneType' and 'datetime.timedelta'
```

### Steps to reproduce
superset load_examples

**Comment by andrewtomai** [comment]
Hi! I would like to attempt to tackle this problem, but am having trouble reproducing the errors.  Is there anything else you did to get these errors, other than:
`$ superset load_examples`

**Comment by mistercrunch** [comment]
It may be fixed on `0.24.0`, `0.23.*` had several bugs.
